internal/backends/mcu/renderer.rs

```diff
@@ -369,21 +372,27 @@ impl PrepareScene {
                 for t in textures.as_slice() {
                     if let Some(dest_rect) = t.rect.intersection(&source_clip).and_then(|r| {
                         r.intersection(
-                            &self.current_state.clip.scale(1. / sx, 1. / sy).round_in().cast(),
+                            &self
+                                .current_state
+                                .clip
+                                .to_untyped()
+                                .scale(1. / sx, 1. / sy)
+                                .round_in()
+                                .cast(),
                         )
                     }) {
                         let actual_x = dest_rect.origin.x - t.rect.origin.x;
                         let actual_y = dest_rect.origin.y - t.rect.origin.y;
                         let stride = t.rect.width() as u16 * bpp(t.format);
                         self.new_scene_texture(
-                            dest_rect.cast().scale(sx, sy),
+                            LogicalRect::from_untyped(&dest_rect.cast().scale(sx, sy)),
```

Comment:
It would be really nice to get rid of this call.

Generally, the intersection with the clip rect is unfortunate. A few lines above we call `to_untyped()` (want to avoid) and here we call `from_untyped` again. For the glyphs we start out in a physical space but have to "temporarily" move to logical space for the interaction and then go back to physical for the `new_scene_texture` call (well, going back is done inside).

Reply:
Also the image size is maybe not strictly speaking physical pixel, but texture pixel or something.
But i don't think that euclid has a type to scale these which is not the same for width and height anyway.
Also we go back to and from logical a bit too much there.  I wonder if it wouldn't be nicer if new_scene_texture takes a pixel in physicial pixel.

Reply:
Yes, I was thinking the same about `new_scene_texture`. I'll give it a try and see if it simplifies the code.